src/main/java/teammates/common/util/AppUrl.java

```diff
@@ -0,0 +1,45 @@
+package teammates.common.util;
+
+/**
+ * The specific implementation of {@link Url} used to encapsulate URLs of the application.
+ * The base URL used in the conversion is the value of app.url in build.properties.
+ */
+public class AppUrl extends Url {
+
+    public AppUrl(String url) {
+        this(Config.APP_URL, url);
+    }
```

Comment:
Instead of this class depending on `Config` class, may be add a factory method somewhere (in the `Config` class?) to create `AppUrl` objects by feeding the `Config.APP_URL` as the base URL?